pkg/scheduler/objects/application_state.go

```diff
@@ -231,6 +239,11 @@ func NewAppState() *fsm.FSM {
 				app := event.Args[0].(*Application) //nolint:errcheck
 				metrics.GetQueueMetrics(app.queuePath).IncQueueApplicationsFailed()
 				metrics.GetSchedulerMetrics().IncTotalApplicationsFailed()
+				if event.Src == Starting.String() {
+					app.queue.decRunningApps()
+					metrics.GetQueueMetrics(app.queuePath).DecQueueApplicationsRunning()
+					metrics.GetSchedulerMetrics().DecTotalApplicationsRunning()
+				}
```

Comment:
We should track this from the Starting state. We are not tracking the metrics in the same way for different state entries or exits. Makes it difficult to comprehend what we do

Reply:
Thanks @wilfred-s,

I just updated the PR,
1. Cleanup resuming
2. Below is all ‘Running Application’ metric changes in application_state. (queue, queue metric, scheduler metric)
   - enter Starting → increment
   - leave Completing → increment if destination is Running
   - leave Starting → decrement, unless destination is Running
   - leave Running → decrement
3. And I found that ‘Accepted Applications’ was incorrect, so I fixed it in the same PR:
   - Was: leave New → increment
   - Is:     leave New → ncrement if destination is not Reject
4. Add test to walk through all nodes.